components/Chips/src/MDCChipView.m

```diff
@@ -604,6 +613,11 @@ - (void)setHighlighted:(BOOL)highlighted {
 }
 
 - (void)setSelected:(BOOL)selected {
+  if (!self.allowsSelection) {
+    // If we disallow selection we don't want to apply any visual or state changes for selection.
+    return;
+  }
```

Comment:
Does this match the uicollectionview behavior? If selection is disabled I believe you can still set the selection to enabled explicitly.

Reply:
Yes, but the allowsSelection is a property on the collection view, not on individual items.
In UICollectionView, selecting a collection view item is independent of selecting a cell. They can be selected or unselected independently of each other.
Here the allowsSelection flag is a chip property, so I think it's weird if you could still select the chip after you set its allowsSelection to false. What do you  think?

Reply:
This looks, then, like a different property than what UICollectionView provides. We may want to name it differently as such because disabling a property from being writeable can result in a variety of surprising behaviors.

Perhaps `shouldSelectOnTap`?

Reply:
Per our discussion, I will remove this condition and rename the API to rippleAllowsSelection.